src/charm.py

```diff
@@ -129,7 +121,7 @@ def _kafka_layer(self) -> Layer:
                     "user": "kafka",
                     "group": "kafka",
                     "environment": {
-                        "KAFKA_OPTS": " ".join(self.kafka_config.extra_args),
+                        "KAFKA_OPTS": " ".join(self.kafka_config.start_args),
```

Comment:
**note:** Not a change for this PR, but I do think we should add a livenessProbe health check that pings something to auto-confirm Kafka is running - and if not - K8s restarts the container. Good self healing feature.